File: src/main/java/com/medical/aesthetic/entity/Customer.java
package com.medical.aesthetic.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "customers")
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Customer extends BaseAuditEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 50)
    private String name;

    @Column(nullable = false, unique = true, length = 20)
    private String phone;

    @Column(length = 20)
    private String idCard;

    private LocalDate birthday;

    @Column(length = 10)
    private String gender;

    @Column(columnDefinition = "TEXT")
    private String medicalHistory;

    @Column(columnDefinition = "TEXT")
    private String allergyHistory;

    @Column(columnDefinition = "TEXT")
    private String remark;
}

File: server/app/models/customer.py
from datetime import datetime

from sqlalchemy import Column, DateTime, Integer, String

from ..database import Base


class Customer(Base):
    __tablename__ = "customers"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(100), nullable=False)
    phone = Column(String(20), index=True, nullable=False)
    id_card = Column(String(20))
    gender = Column(String(10))
    source_channel = Column(String(100))
    remark = Column(String(500))
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)
